File: ruby/hamming/hamming.rb
# Computer the Hamming difference for DNA strands
class Hamming
  def self.compute(dna_a, dna_b)
    unless dna_a.length == dna_b.length
      raise ArgumentError, 'Strands have different length'
    end

    dna_a.chars.zip(dna_b.chars).count { |a, b| a != b }
  end
end

module BookKeeping
  VERSION = 3
end

File: lua/hamming/hamming.lua
local hamming = {}

hamming.compute = function(a, b)
	if a:len() ~= b:len() then
		return -1
	end

	local diff = 0

	for i = 1, #a do
		if a:sub(i, i) ~= b:sub(i, i) then
			diff = diff + 1
		end
	end

	return diff
end

return hamming
